apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx

```diff
@@ -266,11 +271,56 @@ class AddedCoursePane extends PureComponent<AddedCoursePaneProps, AddedCoursePan
         );
     };
 
+    getBarebonesSchedule = () => {
+        const sectionsByTerm = AppStore.getBarebonesSchedule().courses.reduce(
+            (acc: Record<string, string[]>, course) => {
+                if (!acc[course.term]) {
+                    acc[course.term] = [];
+                }
+                acc[course.term].push(course.sectionCode);
+                return acc;
+            }, {});
+
+        return (
+            <>
+                {
+                    //Sections organized under terms, in case the schedule contains multiple terms
+                    Object.entries(sectionsByTerm).map(([term, sections]) => {
+                        return (
+                            <>
+                                <Typography variant="h6">{term}</Typography>
+                                <Paper key={term} elevation={1}>
+                                    <Grid item md={12} xs={12} key={term} style={{padding: "15px 0px 15px"}}>
+                                        <Typography variant="body1">
+                                            Sections enrolled: {sections.join(', ')}
+                                        </Typography>
+                                    </Grid>
+                                </Paper>
+                            </>
+
+                        );
+                    })
+                }
+                <Typography variant="body1">
+                    PeterPortal or WebSoc is currently unreachable.
+                    This is the information that we can currently retrieve.
+                </Typography>
+            </>
+        )
+    }
+
+    barebonesModeChange = () => {
+        this.setState({ barebonesMode: AppStore.getBarebonesMode() }, () => console.log(this.state.barebonesMode));
+    }
+
     render() {
         return (
-            <Grid container spacing={2}>
-                {this.getGrid()}
-            </Grid>
+            this.state.barebonesMode ?
+                this.getBarebonesSchedule()
+                :
+                <Grid container spacing={2}>
+                    {this.getGrid()}
+                </Grid>
```

Comment:
Don't call functions in JSX, this is generally an anti-pattern because they won't be registered by React as components, and so they'll have screwed up state.
**However**, there is a caveat to this rule with class-based components for some reason. But still...don't do it :shrug:

Is this syntax possible at all? Just encapsulate both components entirely as functions, and then render one or the other as a component.

```suggestion
            this.state.barebonesMode ? <this.getBarebonesSchedule /> : <this.getGrid />
```

Reply:
Oh, actually there's two anti patterns lol.
1. Creating a component inside of another component is the main thing that screws up state (unless it's a class component for some reason). Correct practice is to make a standalone component...even if it requires prop-drilling.

2. Using a function call instead of a React JSX tag thingy. This just prevents it from rendering on the component tree properly or something.

TLDR: I love how React is so **easy to use** ™️